### PR TITLE
Post formatter fixes as suggested changes on PRs

### PR DESCRIPTION
**Pull request by chelseakr-exygy**
### Description

When the formatting check fails, the fixed files already exist on the runner: `bin/fmt` and `bin/fmt-sbt` write them to disk, and the workflow diffs and discards them. This adds an `actions/github-script` step that posts that diff back on the PR as inline one-click suggested changes, so authors can apply the fix from the PR page without running anything locally.

Motivating case: two template PRs edited through the web UI (#13933, #13932) failed formatting with no self-serve way out, and each needed a fix commit pushed by someone else. The Slack discussion settled on suggestions rather than auto-committing, so nothing modifies a branch unless the author clicks.

Behavior:

- The check still fails exactly as it does today. The new step only reads the working tree and posts review comments; the existing `git diff --exit-code` step still decides pass/fail.
- Nothing is pushed to the branch, so no PAT or App token is needed and there is no stale-check re-trigger problem. The job gains `pull-requests: write` (it previously opted down to `read-all`).
- Skipped on forked PRs, where the token is read-only.
- Implementation note: the first cut used `reviewdog/action-suggester`, but this repo's Actions policy only allows GitHub-owned actions plus a short allowlist, so the workflow failed at startup. Rather than widening the allowlist, this uses `actions/github-script` at the same pinned version already used elsewhere in this repo. The logic lives in `.github/scripts/post-format-suggestions.js` (the step itself is a two-line `require()`), with unit tests for the diff-parsing edge cases run by a `node --test` workflow step; `bin/fmt` now runs prettier over `.github/scripts/*.js` so the suggester's own code is formatter-covered.
- Known limitation: GitHub suggestions only attach to lines inside the PR's diff. If a whole batch includes such a hunk the step falls back to posting suggestions one at a time and drops only the ones GitHub refuses; the check's own diff output still shows everything.

AI disclosure: the workflow change and this description were drafted with Claude and reviewed by me.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label (see [docs](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-appropriate-labels) for more info): < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [ ] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from `civiform/eng-admin` as FYI (if this PR includes functionality changes and neither you nor the primary reviewer is an admin)
- [ ] Added an additional reviewer as required if changes potentially affect the security of the application.
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [ ] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [ ] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`
- [x] Noted in the PR description which, if any, code in this PR was generated by AI.

### Instructions for manual testing

Already demonstrated end-to-end on this PR: the `DEMO ONLY` commit deliberately misformatted `admin_export_view.ts`, the Format run posted the fixes as one-click suggestions while the check still failed as usual, and applying them via "Commit suggestion" restored the file to match main and turned the check green — so the revert itself was performed by the feature under review. (An earlier iteration used `-U1` diff context, which produced suggestions GitHub refused to apply because their ranges included deleted lines; the `-U0` anchoring in the final version is the fix.) The demo and suggestion commits disappear at squash merge.

### CODE REVIEW

File: .github/scripts/post-format-suggestions.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+// wider range sweeps in the PR diff's deleted lines, and GitHub refuses
+// to apply suggestions whose range includes deleted lines.
+function parseDiff(diff) {
+  const files = []
+  let file = null
+  let inHeader = false
+  for (const line of diff.split('\n')) {
+    // With -U0 every content line starts with '+', '-', or '\', so a
+    // bare 'diff --git ' line is always a real file boundary.
+    // '+++ b/...' alone is not: an added content line beginning
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js]: **nitpick:** The `:` is odd phrasing here, it's hard to tell what its conveying.

Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js]: **suggestion:** 
```suggestion
    // '+++ b/...' alone is not a file boundary because an added content line beginning
```

File: .github/scripts/post-format-suggestions.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    // '++ b/...' renders exactly the same way, so header lines only
+    // count between 'diff --git ' and the file's first '@@ '.
+    if (line.startsWith('diff --git ')) {
+      inHeader = true
+      file = null
+    } else if (inHeader && line.startsWith('+++ b/')) {
+      // Headers for paths with special characters arrive quoted
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js:38]: **suggestion:** This could use a guidepost for what it's doing, which is grabbing the filename.  The comment largely described why it's tricky and not what it's doing.

File: .github/scripts/post-format-suggestions.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      file = {path: line.slice(6).replace(/\t$/, ''), hunks: []}
+      files.push(file)
+    } else if (line.startsWith('@@ ')) {
+      inHeader = false
+      if (file) {
+        const match = line.match(/^@@ -(\d+)(?:,(\d+))? /)
+        file.hunks.push({
+          start: Number(match[1]),
+          count: match[2] === undefined ? 1 : Number(match[2]),
+          lines: [],
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js]: **suggestion:** `newLines` maybe for symmetry and clarity.

File: .github/scripts/post-format-suggestions.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+// rejects the whole review.
+const MAX_BODY_LENGTH = 65536
+
+// The review API caps out well before this, and a wall of suggestions
+// stops being helpful anyway; the review body discloses the rest.
+const MAX_COMMENTS = 50
+
+// Parses `git diff -U0` output into {path, hunks} entries, where each
+// hunk carries its old-file start line and count plus the replacement
+// (lines) and replaced (oldLines) content. Zero context lines matter:
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js]: **question:** The inclusion of "zero context" and "wider range sweeps" is a little hard to follow with the current context.  Can those be clarified?

Is `zero context lines` when there is no recognized prefix?

File: .github/scripts/post-format-suggestions.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    for (const {start, count, lines, oldLines} of hunks) {
+      // A pure insertion has no existing line to anchor on, and a hunk
+      // whose only change is the trailing-newline marker yields a
+      // suggestion identical to the existing text, which GitHub can't
+      // apply.
+      if (count === 0 || lines.join('\n') === oldLines.join('\n')) {
+        skipped++
+        continue
+      }
+      // The fence must be longer than any backtick run in the content,
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js:83]: **suggestion:** Another place a guidepost would be useful

"Ensure the final suggestion including the suggestion markdown fencing isn't too long"

File: .github/scripts/post-format-suggestions.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+      )
+      const body =
+        fence + 'suggestion\n' + lines.map((l) => l + '\n').join('') + fence
+      if (body.length > MAX_BODY_LENGTH) {
+        skipped++
+        continue
+      }
+      comments.push({
+        path,
+        line: start + count - 1,
```
Comment (shane-exygy) [inline .github/scripts/post-format-suggestions.js:106]: **suggestion:** This could use a comment, it's hard to infer why line should be different than start, and the docs are readily accessible from this file

**Review by shane-exygy — CHANGES_REQUESTED** [review]

**Comment by chelseakr-exygy** [comment]
@shane-exygy tysm for the review! zero context meant the -U0 flag, not the prefixes - it tells git to print no unchanged lines around a hunk, so each hunk covers exactly the lines the formatter changed and the suggestion anchors on just those. with any context lines the range gets wider and can sweep in lines this PR deletes, and github won't apply a suggestion whose range includes a deleted line. reworded the comment in to say that directly instead of hinting at it!

File: .github/scripts/post-format-suggestions.js (hunk 4 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js]: zero context meant the -U0 flag, not the prefixes - it tells git to print no unchanged lines around a hunk, so each hunk covers exactly the lines the formatter changed and the suggestion anchors on just those. with any context lines the range gets wider and can sweep in lines this PR deletes, and github won't apply a suggestion whose range includes a deleted line. reworded the comment in 7ffbc63 to say that directly instead of hinting at it

File: .github/scripts/post-format-suggestions.js (hunk 1 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js]: applied your wording in 7ffbc63, just rewrapped to the comment width

File: .github/scripts/post-format-suggestions.js (hunk 2 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js:38]: done in 7ffbc63

File: .github/scripts/post-format-suggestions.js (hunk 3 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js]: renamed in 7ffbc63

File: .github/scripts/post-format-suggestions.js (hunk 5 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js:83]: done in 7ffbc63

File: .github/scripts/post-format-suggestions.js (hunk 6 above)
Comment (chelseakr-exygy) [reply .github/scripts/post-format-suggestions.js:106]: added a comment with the docs link in 7ffbc63

Comment (shane-exygy) [reply .github/scripts/post-format-suggestions.js:106]: oh wow, thanks for the comment, that is an unintuitive name

**Review by shane-exygy — APPROVED** [review]
